shuup/admin/modules/products/forms/base_forms.py

```diff
@@ -202,41 +204,65 @@ def __init__(self, **kwargs):
         self.fields["shipping_methods"].queryset = shipping_methods_qs
         self.fields["default_price_value"].required = True
 
-        initial_suppliers = []
         initial_categories = []
+        initial_suppliers = []
 
         if self.instance.pk:
-            initial_suppliers = self.instance.suppliers.all()
             initial_categories = self.instance.categories.all()
+            initial_suppliers = self.instance.suppliers.all()
         elif not settings.SHUUP_ENABLE_MULTIPLE_SUPPLIERS:
             supplier = Supplier.objects.first()
             initial_suppliers = ([supplier] if supplier else [])
 
-        self.fields["suppliers"] = Select2ModelMultipleField(
-            initial=initial_suppliers,
-            model=Supplier,
-            widget=QuickAddSupplierMultiSelect(
+        if settings.SHUUP_ADMIN_LOAD_SELECT_OBJECTS_ASYNC.get("suppliers"):
+            self.fields["suppliers"] = Select2ModelMultipleField(
                 initial=initial_suppliers,
-                attrs={"data-search-mode": "enabled"}
-            ),
-            required=False
-        )
-        self.fields["primary_category"] = Select2ModelField(
-            initial=(self.instance.primary_category if self.instance.pk else None),
-            model=Category,
-            widget=QuickAddCategorySelect(
-                editable_model="shuup.Category",
+                model=Supplier,
+                widget=QuickAddSupplierMultiSelect(
+                    initial=initial_suppliers,
+                    attrs={"data-search-mode": "enabled"}
+                ),
+                required=False
+            )
+        else:
+            self.fields["suppliers"].widget = QuickAddSupplierMultiSelect(initial=initial_suppliers)
+
+        if settings.SHUUP_ADMIN_LOAD_SELECT_OBJECTS_ASYNC.get("categories"):
+            self.fields["primary_category"] = Select2ModelField(
                 initial=(self.instance.primary_category if self.instance.pk else None),
-                attrs={"data-placeholder": ugettext("Select a category")}
-            ),
-            required=False
-        )
-        self.fields["categories"] = Select2ModelMultipleField(
-            initial=initial_categories,
-            model=Category,
-            widget=QuickAddCategoryMultiSelect(initial=initial_categories),
-            required=False
-        )
+                model=Category,
+                widget=QuickAddCategorySelect(
+                    editable_model="shuup.Category",
+                    initial=(self.instance.primary_category if self.instance.pk else None),
```

Comment:
This looks like taking iterable.

Reply:
Iterable? It was like this before. This is a `Select2ModelField` not a multiple field. :thinking: